spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java

```diff
@@ -47,4 +47,8 @@ private SparkSQLProperties() {}
   public static final String PRESERVE_DATA_GROUPING =
       "spark.sql.iceberg.planning.preserve-data-grouping";
   public static final boolean PRESERVE_DATA_GROUPING_DEFAULT = false;
+
+  // Controls how many physical file deletes to execute in parallel when not otherwise specified
+  public static final String DELETE_PARALLELISM = "driver-delete-default-parallelism";
+  public static final String DELETE_PARALLELISM_DEFAULT = "25";
```

Comment:
With S3's request throttling around 4k requests a second this gives us a lot of overhead. 
Assuming a 50ms response time
4000 max requests / Second / 20 requests per thread per second =~  200 max concurrent requests. 

Another option for this is to also incorporate the "bulk delete" apis but that would only help with S3 based filesystems.